packages/compass-components/src/components/drawer-portal.tsx

```diff
@@ -116,18 +132,28 @@ export const DrawerContentProvider: React.FunctionComponent = ({
 
   return (
     <DrawerStateContext.Provider value={drawerState}>
-      <DrawerActionsContext.Provider value={drawerActions}>
-        {children}
-      </DrawerActionsContext.Provider>
+      <DrawerOpenStateContext.Provider value={drawerOpenState}>
+        <DrawerSetOpenStateContext.Provider
+          value={{ setIsDrawerOpen: setDrawerOpenState }}
```

Comment:
Otherwise you're creating a different version of value every render leading to everything depending on this context to re-render

```suggestion
          value={setDrawerOpenState}
```

Reply:
Ah right 🤦 Thank you!